dev/devicelab/bin/tasks/flutter_gallery__back_button_memory.dart

```diff
@@ -2,11 +2,49 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+/// Measure application memory usage after pausing and resuming the app
+/// with the Android back button.
+
 import 'dart:async';
 
-import 'package:flutter_devicelab/tasks/perf_tests.dart';
+import 'package:flutter_devicelab/framework/adb.dart';
 import 'package:flutter_devicelab/framework/framework.dart';
+import 'package:flutter_devicelab/framework/utils.dart';
+import 'package:flutter_devicelab/tasks/perf_tests.dart';
+
+const String packageName = 'io.flutter.demo.gallery';
+const String activityName = 'io.flutter.demo.gallery.MainActivity';
+
+class BackButtonMemoryTest extends MemoryTest {
+  BackButtonMemoryTest() : super('${flutterDirectory.path}/examples/flutter_gallery', 'test_memory/back_button.dart', packageName);
+
+  @override
+  AndroidDevice get device => super.device;
+
+  /// Perform a series of back button suspend and resume cycles.
+  @override
+  Future<void> useMemory() async {
+    await launchApp();
+    await recordStart();
+    for (int iteration = 0; iteration < 10; iteration += 1) {
+      print('back/forward iteration $iteration');
+      prepareForNextMessage('AppLifecycleState.paused');
+      await device.shellExec('input', <String>['keyevent', 'KEYCODE_BACK']);
+      await ready;
+      await new Future<Null>.delayed(const Duration(milliseconds: 100));
```

Comment:
Why do we need this delay? (I'm sure there's a reason, maybe add a comment to explain it?)

Reply:
added comments